File: Task/Sum-digits-of-an-integer/Standard-ML/sum-digits-of-an-integer.ml
fun sumDigits (0, _) = 0
  | sumDigits (n, base) = n mod base + sumDigits (n div base, base)

val testInput = [(1, 10), (1234, 10), (0xfe, 16), (0xf0e, 16)]
val () = print (String.concatWith " " (map (Int.toString o sumDigits) testInput) ^ "\n")

File: Task/Sum-digits-of-an-integer/Zen-C/sum-digits-of-an-integer.c
fn digits_sum(n: int, b: const int) -> int {
    let sum = 0;
    while n > 0 {
        sum += n % b;
        n /= b;
    }
    return sum;
}

fn main() {
    let a: (int, int)[4] = [(1, 10), (1234, 10), (0xfe, 16), (0xf0e, 16)];
    for i in 0..a.len {
        let (n, b) = a[i];
        println "{digits_sum(n, b)}";
    }
}
